packages/api-fetch/src/middlewares/fetch-all-middleware.js

```diff
@@ -91,7 +92,7 @@ const fetchAllMiddleware = async ( options, next ) => {
 		} ),
 		// Ensure headers are returned for page 1.
 		parse: false,
-	} );
+	} ).catch( parseAndThrowError );
```

Comment:
Notice that `parse` option will be always be `true` here because it skips earlier in the first conditional of the `fetchAllMiddleware`.